src/libraries/System.Private.CoreLib/src/System/IO/FileSystemInfo.cs

```diff
@@ -107,6 +115,46 @@ public DateTime LastWriteTimeUtc
             set => LastWriteTimeCore = File.GetUtcDateTimeOffset(value);
         }
 
+        /// <summary>
+        /// If this <see cref="FileSystemInfo"/> instance represents a link, returns the link target's path.
+        /// If a link does not exist in <see cref="FullName"/>, or this instance does not represent a link, returns <see langword="null"/>.
+        /// </summary>
+        public string? LinkTarget => _linkTarget ??= FileSystem.GetLinkTarget(FullPath, this is DirectoryInfo);
+
+        /// <summary>
```

Comment:
Note: when `FileSystem.GetLinkTarget` returns `null`, next time `LinkTarget` is called, `FileSystem.GetLinkTarget` will called again. The other properties on `FileSystemInfo` return a cached value until `Refresh` is called.

Reply:
While this will help us to achieve consistency with `IsLink` in the future, I still have the concern about _caching null_ in this property. @adamsitnik thoughts?

Reply:
The properties on this object are cached until `Refresh`. If `LinkTarget` doesn't follow this behavior, maybe it should have been a method.

> While this will help us to achieve consistency with IsLink in the future

See https://github.com/dotnet/runtime/pull/54253#discussion_r665121768.